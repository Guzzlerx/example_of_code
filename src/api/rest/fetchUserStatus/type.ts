import { UserStatusesEnum } from 'constants/userStatuses/userStatuses';

export interface IUserStatusResponse {
  email: string;
  clientStatus: UserStatusesEnum;
}
