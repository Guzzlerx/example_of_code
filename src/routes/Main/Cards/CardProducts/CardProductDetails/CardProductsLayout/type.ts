import { CURRENCY_CODES_ENUM, PAYMENT_SYSTEMS_ENUM } from 'constants/cards/cards';
import { ICardProductAdvantagesProps } from '../CardProductDetails/Parts/CardProductAdvantages/type';

export interface ICardProductData {
  cardName: ICardProductAdvantagesProps['cardName'];
  currency: CURRENCY_CODES_ENUM;
  paymentSystem: PAYMENT_SYSTEMS_ENUM;
}

export interface IUseCardProductContext extends ICardProductData {
  showCardInfo: ({ cardName, currency, paymentSystem }: ICardProductData) => void;
}
