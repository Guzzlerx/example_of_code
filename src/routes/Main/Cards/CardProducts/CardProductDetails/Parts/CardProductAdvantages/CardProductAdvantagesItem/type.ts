import { CARD_NAMES_ENUM } from 'constants/cards/cards';

export interface ICardProductAdvantagesItemProps {
  cardName: Exclude<CARD_NAMES_ENUM, CARD_NAMES_ENUM.PERSONAL_LOAN>;
}
