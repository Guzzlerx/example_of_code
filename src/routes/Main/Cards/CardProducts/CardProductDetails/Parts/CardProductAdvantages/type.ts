import { CARD_NAMES_ENUM } from 'constants/cards/cards';
import { ICardConfigsAdvantages } from 'constants/cards/types';

export interface ICardProductAdvantagesProps {
  cardName: Exclude<CARD_NAMES_ENUM, CARD_NAMES_ENUM.PERSONAL_LOAN>;
  data: ICardConfigsAdvantages[];
}
