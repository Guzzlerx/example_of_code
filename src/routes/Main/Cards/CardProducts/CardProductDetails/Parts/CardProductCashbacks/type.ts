import { CARD_NAMES_ENUM } from 'constants/cards/cards';

export interface ICardProductCashbacksProps {
  cardName: Exclude<
    CARD_NAMES_ENUM,
    CARD_NAMES_ENUM.SMART_PAYOUT | CARD_NAMES_ENUM.INFINITY_PREMIUM | CARD_NAMES_ENUM.PERSONAL_LOAN
  >;
}
