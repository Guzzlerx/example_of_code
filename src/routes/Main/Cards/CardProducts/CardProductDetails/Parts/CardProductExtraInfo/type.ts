import { CARD_NAMES_ENUM } from 'constants/cards/cards';

export interface ICardProductExtraInfoProps {
  cardName: Extract<
    CARD_NAMES_ENUM,
    CARD_NAMES_ENUM.INFINITY_PREMIUM | CARD_NAMES_ENUM.SMART_PAYOUT
  >;
}
