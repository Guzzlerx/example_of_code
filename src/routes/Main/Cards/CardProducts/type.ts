import { CURRENCY_CODES_ENUM, PAYMENT_SYSTEMS_ENUM } from 'constants/cards/cards';
import { ICardProductAdvantagesProps } from './CardProductDetails/Parts/CardProductAdvantages/type';

export interface IMyCardProductItem {
  id: string;
  currencyCode: CURRENCY_CODES_ENUM;
  paymentSystem: PAYMENT_SYSTEMS_ENUM;
  cardName: ICardProductAdvantagesProps['cardName'];
}
