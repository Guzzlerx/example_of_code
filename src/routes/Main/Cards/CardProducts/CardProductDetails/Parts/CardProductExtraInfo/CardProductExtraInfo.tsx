import { ReactNode } from 'react';
import { AFlexContainer } from 'components/AFlexContainer/AFlexContainer';
import { ReactComponent as ChatBotImage } from 'assets/image/cardProductExtraInfo/chat_bot.svg';
import { ReactComponent as CryptoFlowersImage } from 'assets/image/cardProductExtraInfo/crypto_flowers.svg';
import { CARD_NAMES_ENUM } from 'constants/cards/cards';
import { ExtraInfoContent } from './ExtraInfoContent/ExtraInfoContent';
import { ICardProductExtraInfoProps } from './type';

export const CardProductExtraInfo = ({ cardName }: ICardProductExtraInfoProps) => {
  let image: ReactNode;

  switch (cardName) {
    case CARD_NAMES_ENUM.INFINITY_PREMIUM:
      image = <ChatBotImage data-testid="chatBotImage" />;
      break;

    case CARD_NAMES_ENUM.SMART_PAYOUT:
      image = <CryptoFlowersImage data-testid="cryptoFlowersImage" />;
      break;

    default:
      image = null;
  }

  return image ? (
    <AFlexContainer data-testid="CardProductExtraInfo" alignItems="center">
      {image}
      <ExtraInfoContent cardName={cardName} />
    </AFlexContainer>
  ) : null;
};
