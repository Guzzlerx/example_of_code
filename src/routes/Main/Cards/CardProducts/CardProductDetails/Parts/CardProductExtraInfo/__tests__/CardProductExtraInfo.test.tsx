import { RenderResult, render, screen } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { CARD_NAMES_ENUM } from 'constants/cards/cards';
import i18n from 'utils/languages/i18nForTests';
import { CardProductExtraInfo } from '../CardProductExtraInfo';
import { ICardProductExtraInfoProps } from '../type';

describe('CardProductExtraInfo', () => {
  let renderResult: RenderResult;

  const props: ICardProductExtraInfoProps = {
    cardName: CARD_NAMES_ENUM.SMART_PAYOUT,
  };

  const props2: ICardProductExtraInfoProps = {
    cardName: CARD_NAMES_ENUM.INFINITY_PREMIUM,
  };

  beforeEach(() => {
    renderResult = render(
      <I18nextProvider i18n={i18n}>
        <CardProductExtraInfo {...props} />
      </I18nextProvider>,
    );
  });

  it('should render correctly', () => {
    expect(renderResult.container).toBeInTheDocument();
  });

  it('snapshot', () => {
    expect(renderResult.container).toMatchSnapshot();
  });

  it('should render the correct image for SMART_PAYOUT', () => {
    const element1 = screen.getByTestId('cryptoFlowersImage');
    expect(element1).toBeInTheDocument();
  });

  it('should render the correct image for INFINITY_PREMIUM', () => {
    renderResult.rerender(
      <I18nextProvider i18n={i18n}>
        <CardProductExtraInfo {...props2} />
      </I18nextProvider>,
    );

    const element2 = screen.getByTestId('chatBotImage');
    expect(element2).toBeInTheDocument();
  });
});
