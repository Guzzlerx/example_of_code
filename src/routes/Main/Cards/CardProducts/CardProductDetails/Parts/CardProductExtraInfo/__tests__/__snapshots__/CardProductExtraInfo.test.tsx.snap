// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardProductExtraInfo snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe iRpbZk"
    data-testid="CardProductExtraInfo"
  >
    <svg
      data-testid="cryptoFlowersImage"
    >
      crypto_flowers.svg
    </svg>
    <div
      class="sc-hLBbgP eyZLuc"
    >
      <h2
        class="sc-gswNZR ePspNt"
        color="#212529"
        size="20"
      >
        Searching for a new way of getting money?
      </h2>
      <h2
        class="sc-gswNZR ewsImF sc-dkrFOg bxiwyQ"
        size="14"
      >
        Thanks to a new program, provided by our bank, you have a possibility to open a deposit account in any type of crypto. This is what you get:
      </h2>
      <ul
        class="sc-eDvSVe gBCMGc"
      >
        <li
          class="sc-jSUZER kQeRHy"
        >
          Higher interest rate. Get up to 
          <strong>
            7% APR
          </strong>
          , 2 times more than with usual deposit;
        </li>
        <li
          class="sc-jSUZER kQeRHy"
        >
          Diversification. Adding crypto savings to your portfolio can help mitigate risk;
        </li>
        <li
          class="sc-jSUZER kQeRHy"
        >
          Security. Cryptocurrencies are stored on decentralized blockchain networks, which are highly secure and resistant to hacking.
        </li>
      </ul>
    </div>
  </div>
</div>
`;
